cli/dcoscli/task/main.py

```diff
@@ -181,20 +181,31 @@ def _ls(task, path, long_, completed):
         path = path[1:]
 
     dcos_client = mesos.DCOSClient()
-    task_obj = mesos.get_master(dcos_client).task(
-                   fltr=task, completed=completed)
-    dir_ = posixpath.join(task_obj.directory(), path)
+    task_objects = mesos.get_master(dcos_client).tasks(
+        fltr=task,
+        completed=completed)
+
+    if len(task_objects) == 0:
+        raise DCOSException(
+            'Cannot find a task with ID containing "{}"'.format(task))
```

Comment:
this is tested here: https://github.com/dcos/dcos-cli/blob/master/cli/tests/integrations/test_task.py#L274